src/fmt.jl

```diff
@@ -62,10 +73,21 @@ end
 # methods to get the current default objects
 # note: if you want to set a default for an abstract type (i.e. AbstractFloat)
 # you'll need to extend this method like here:
+
+const ComplexInteger  = Complex{<:Integer}
+const ComplexFloat    = Complex{<:AbstractFloat}
+const ComplexRational = Complex{<:Rational}
+
 default_spec(::Type{<:Integer})        = DEFAULT_FORMATTERS[Integer]
 default_spec(::Type{<:AbstractFloat})  = DEFAULT_FORMATTERS[AbstractFloat]
 default_spec(::Type{<:AbstractString}) = DEFAULT_FORMATTERS[AbstractString]
 default_spec(::Type{<:AbstractChar})   = DEFAULT_FORMATTERS[AbstractChar]
+default_spec(::Type{<:AbstractIrrational}) = DEFAULT_FORMATTERS[AbstractIrrational]
+default_spec(::Type{<:Rational})       = DEFAULT_FORMATTERS[Rational]
+default_spec(::Type{<:Number})         = DEFAULT_FORMATTERS[Number]
+default_spec(::ComplexInteger)         = DEFAULT_FORMATTERS[ComplexInteger]
```

Comment:
Again, you are confusing passing an instance of a particular type, with passing the type itself.

You need to have `::Type{<:Complex}`.

I think it's best to work on making any complex type work at first, before trying to add the capability of having separate defaults for different parameterized Complex types.